frontend/src/components/data-table/column-pinning/feature.ts

```diff
@@ -0,0 +1,194 @@
+    return {
+      // default to no pinning
+      columnPinning: { left: [], right: [] },
+      ...state,
+    };
+  },
+
+  // define the column pinning's default options
+  getDefaultOptions: <TData extends RowData>(
+    table: Table<TData>,
+  ): ColumnPinningOptions => {
+    return {
+      enableColumnPinning: true,
+      onColumnPinningChange: makeStateUpdater("columnPinning", table),
+    } as ColumnPinningOptions;
+  },
+
+  createColumn: <TData extends RowData>(
+    column: Column<TData>,
+    table: Table<TData>,
+  ) => {
+    // get the column's pinning state
+    column.getIsPinned = (): ColumnPinningPosition => {
+      const pinningState = table.getState().columnPinning;
+      return pinningState.left?.includes(column.id)
+        ? "left"
+        : pinningState.right?.includes(column.id)
+          ? "right"
+          : false;
+    };
+
+    // get if the column can be pinned
+    column.getCanPin = (): boolean => {
+      return table.options.enableColumnPinning ?? false;
+    };
+
+    // set the column's pinning state
+    column.setColumnPinning = (position?: ColumnPinningPosition) => {
+      const safeUpdater: Updater<ColumnPinningState> = (old) => {
+        // get the previous pinning state
+        const prevValue = old.left?.includes(column.id)
+          ? "left"
+          : old.right?.includes(column.id)
+            ? "right"
+            : false;
+
+        // initialize the new pinning state
+        let newState: ColumnPinningState = old;
+
+        // If prevValue and position are equal, do nothing
+        if (prevValue === position) {
+          return newState;
+        }
+
+        // If prevValue is false and position is set,
+        // add the column to the specified side
+        if (prevValue === false && position) {
+          newState = {
+            ...old,
+            [position]: [...(old[position] || []), column.id],
+          };
+        }
+
+        // If prevValue isn't false and position is false,
+        // remove the column from the specified side
+        if (prevValue !== false && !position) {
+          newState = {
+            ...old,
+            [prevValue]: old[prevValue]?.filter((id) => id !== column.id),
+          };
+        }
+
+        // If prevValue isn't false and position is different,
+        // move the column to the new side
+        if (prevValue !== false && position && prevValue !== position) {
+          newState = {
+            ...old,
+            [prevValue]: old[prevValue]?.filter((id) => id !== column.id),
+            [position]: [...(old[position] || []), column.id],
+          };
+        }
+
+        // If only "select" column is pinned to the left, unpin it
+        if (
+          position !== "left" &&
+          newState.left?.length === 1 &&
+          newState.left[0] === "select"
+        ) {
+          // unpin the 'select' column
+          table.getColumn("select")?.pin(false);
```

Comment:
can we move `"select"` to a constant to be shared between here and the `SelectColumn`. maybe we can rename to "__select__" so its more unique

Reply:
Do you mean to rename `"select"` to `"__select__"` here

https://github.com/marimo-team/marimo/blob/9a7304dd569a7b307db56308129edf29a9c129cc/frontend/src/components/data-table/columns.tsx#L151

Reply:
yea i am realizing that would be better since a column could be named `"select"` coming from a dataframe